### ISSUE TITLE
having a problem when using --unpaired

### ISSUE THREAD

**Issue by PiraMew** [mossmatters/HybPiper#59]
Hello, I am  a PhD student working with Wolf Eiserhardt, and I am trying to run reads_first.py but get the error when using --unpaired flag
Here is what I get;

```
##Extracted from where the error occure
Running SPAdes on 352 genes
time parallel --eta spades.py --only-assembler --threads 1 --cov-cutoff 8 -s {}/{}_unpaired.fasta --12 {}/{}_interleaved.fasta -o {}/{}_spades :::: spades_genelist.txt > spades.logComputers / CPU cores / Max jobs to run
1:local / 24 / 24Computer:jobs running/jobs completed/%of started jobs/Average seconds to complete
ETA: 0s Left: 0 AVG: 0.33s  local:0/352/100%/0.3s s   real	1m57.343s
user	4m54.333s
sys	0m35.985s
ERROR: One or more genes had an error with SPAdes assembly. This may be due to low coverage. No contigs found for the following genes:
6933
6563
5968
5865
.
.
.
## Leading to this warning
WARNING: All Kmers failed for 6933!
WARNING: All Kmers failed for 6563!
WARNING: All Kmers failed for 5968!
WARNING: All Kmers failed for 5865!
WARNING: All Kmers failed for 6029!
.
.
.
## This made the re-running SPAdes not be initiated
Re-running SPAdes for 0 genes
parallel --eta --timeout 400% :::: redo_spades_commands.txt > spades_redo.logComputers / CPU cores / Max jobs to run
1:local / 24 / 1
ETA: 0s Left: 0 AVG: 0.00s  0
All redos completed successfully!
Running Exonerate to generate sequences for 6 genes
time parallel --eta python /home/psumanon/HybPiper/exonerate_hits.py {}/{}_baits.fasta {}/{}_contigs.fasta --prefix {}/540AL1_Maesapalauensis -t 65 --depth_multiplier 10 --length_pct 90 :::: exonerate_genelist.txt > genes_with_seqs.txt

```
when I checked spades.log, I found this message:
`
== Error ==  file is empty: /faststorage/project/Maesa/steps/HybPip/testunpaired/test/540AL1_Maesapalauensis/5343/5343_unpaired.fasta (single reads, library number: 1, library type: paired-end)
`
There are files `<gene>_unpaired.fasta` in the sample output directory, but no content in it. 
My inputs `<sample>_unpaired.fasta` also have reads in it.

On the contrary, when not using --unpaired, the re-running of SPAdes was processed and resulted in more genes recovered. It also showed the first error message about low coverage, but the redo SPAdes was performed.

I am running the program on a cluster which all dependencies were install using conda. 
However, I think it is not a cluster's issue since when I tried running it on my own computer with the same installation method, I also got that error.

Below is the script I used. I run it on /scratch/ space of the cluster,
```
#!/bin/bash
#SBATCH --account=xxx
#SBATCH --job-name=readfirst
#SBATCH --time=07-00:00:00
#SBATCH --mem-per-cpu=30G
#SBATCH --cpus-per-task=32

# Create a temporary directory for the job in local storage
TMPDIR=/scratch/$SLURM_JOBID
export TMPDIR
mkdir -p $TMPDIR
cd $TMPDIR

# Copy input file(s) to local storage if necessary
cp <path/to/trimmed/fastq>/*.fastq .

#create namelist first
for f in *_1_paired_trimmed.fastq; do (echo ${f/_1_paired_trimmed.fastq} >> namelist.txt); done

#run reads_first.py
while read name; 
do /home/psumanon/HybPiper/reads_first.py -b /home/psumanon/target/mega353.fasta -r ${name}_1_paired_trimmed.fastq ${name}_2_paired_trimmed.fastq --unpaired ${name}_unpaired.fastq --prefix $name --bwa
done < namelist.txt


# Move the result files, logs, and any relevant data back to user's home directory. 
mkdir -p $SLURM_SUBMIT_DIR/$SLURM_JOBID
cp -r * $SLURM_SUBMIT_DIR/$SLURM_JOBID/
```

**Comment by mossmatters** [comment]
My first guess would be that the redo script is not accounting for the `--unpaired` flag. What is the content of the `redo_spades_commands.txt` file? The commands in that file should be restarting spades runs that failed with a segmentation fault, so either the file is not being written to when using the unpaired flag, or the commands in that file are incorrect.

Would you be willing to share your three FASTQ files and the target file, so I can check what is happening on my end? You can e-mail me directly (mossmatters@gmail.com) if you don't want to post the link here.

**Comment by PiraMew** [comment]
Hello Matt, there is no content in the redo_spades_commands.txt file.
I have sent you the files that you asked via e-mail.

**Comment by PiraMew** [comment]
Hello again @mossmatters, do you think is it possible that the cause of the problem is related to running the program in conda environment?
I did two testing runs on my own computer, in conda-created environment and in base environment.
The run in the conda environment got the same error, while one in base environment accomplished the task without that error.

**Comment by chrisjackson-pellicle** [comment]
Hi @PiraMew,

This was a bug in HybPiper - it should now be resolved in version 2.0.1 (see #54).

Cheers,

Chris
